docs/src/components/copy-button.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { CheckIcon, CopyIcon } from "@radix-ui/react-icons";
+
+import { cn } from "@/lib/utils";
+import { Button } from "@/components/ui/button";
+
+interface CopyButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  value: string;
+  src?: string;
+}
+
+async function copyToClipboard(value: string) {
+  navigator.clipboard.writeText(value);
+}
+
+export function CopyButton({
+  value,
+  className,
+  src,
+  ...props
+}: CopyButtonProps) {
+  const [hasCopied, setHasCopied] = React.useState(false);
+
+  React.useEffect(() => {
+    setTimeout(() => {
+      setHasCopied(false);
+    }, 2000);
+  }, [hasCopied]);
+
+  return (
+    <Button
+      size="icon"
+      variant="ghost"
+      className={cn(
+        "relative z-10 h-6 w-6 text-zinc-50 hover:bg-zinc-700 hover:text-zinc-50",
+        className,
+      )}
+      onClick={() => {
+        copyToClipboard(value);
+        setHasCopied(true);
+      }}
```

Comment:
I don't know how astro specifically manages re-renders and state, but when sending arrow functions to react components it's usually recommended to wrap them with a useCallback to avoid any accidental object comparison issues in effects or re-renders.

Reply:
Astro doesn't have anything to do with managing React renders and state afaik